### PR TITLE
Feature/camera controls

### PR DESCRIPTION
**Pull request by interim17**
Getting started adding UI components to Simularium in Jupyter Notebooks.

Added an icons file, and we are grabbing TrajectoryData and parsing it to send down props.

Otherwise it's a pretty close copy of the JSX and CSS for the camera controls from website repo.

### CODE REVIEW

**Comment by ShrimpCryptid** [comment]
Is the build failing an issue?

File: css/CameraControls.css
Comment (ShrimpCryptid) [inline css/CameraControls.css:1]: Probably out of scope of this PR, but does it make sense to use something like CSS modules to prevent classname conflicts? Or keeping the CSS in the same directory as the component? This project seems to be structured differently from some of our other ones.

File: src/Viewer.tsx [hunk 1]
```diff
@@ -47,7 +49,7 @@ function ViewerWidget(props: WidgetProps): JSX.Element {
         backgroundColor={[0, 0, 0]}
         height={props.height}
         width={props.width}
-        loggerLevel="debug"
+        loggerLevel=""
```
Comment (ShrimpCryptid) [inline src/Viewer.tsx]: Should this property be removed instead of left blank?

**Review by ShrimpCryptid — APPROVED** [review]
Tested locally and things seemed to be working! Opened a separate PR #4 for a few issues I encountered while setting up the dev environment.

**Comment by ascibisz** [comment]
> Is the build failing an issue?

Yep, we should get that fixed!

```
src/Viewer.tsx(58,9): error TS2741: Property 'colorChanges' is missing in type '{ highlightedAgents: never[]; hiddenAgents: never[]; }' but required in type 'SelectionStateInfo'.
```
Based on this error message, it seems like that's an artifact of the changes Joe and Megan made in the simularium-viewer recently that are now merged! Seems like a simple change to add that property in `Viewer.tsx`. @interim17 if you could handle that in this PR, that'd be great!

File: src/Viewer.tsx (hunk 1 above)
Comment (interim17) [reply src/Viewer.tsx]: I'm not sure! When set to debug it renders a green box on the canvas which was annoying to look at so I set it to the empty string. loggerLevel is not an optional property so I think it needs to be assigned some value.

File: css/CameraControls.css
Comment (interim17) [reply css/CameraControls.css:1]: Last I heard from @meganrm we were not going to use modules, but I don't think that's set in stone.

I'm open about where to keep the stylesheets, I just stuck them in the css folder Megan had already set up. Sometimes having unique class names can make inspecting the elements a bit easier actually so there aren't many html elements with the classname `container` etc.

**Comment by interim17** [comment]
> > Is the build failing an issue?
> 
> Yep, we should get that fixed!
> 
> ```
> src/Viewer.tsx(58,9): error TS2741: Property 'colorChanges' is missing in type '{ highlightedAgents: never[]; hiddenAgents: never[]; }' but required in type 'SelectionStateInfo'.
> ```
> 
> Based on this error message, it seems like that's an artifact of the changes Joe and Megan made in the simularium-viewer recently that are now merged! Seems like a simple change to add that property in `Viewer.tsx`. @interim17 if you could handle that in this PR, that'd be great!

DIdn't anticipate that because I hadn't update the viewer locally!  @ascibisz  pushing the change you suggested fixed the mac build but not linux. I'll keep poking around on that one.

**Comment by interim17** [comment]
> > > Is the build failing an issue?
> > 
> > 
> > Yep, we should get that fixed!
> > ```
> > src/Viewer.tsx(58,9): error TS2741: Property 'colorChanges' is missing in type '{ highlightedAgents: never[]; hiddenAgents: never[]; }' but required in type 'SelectionStateInfo'.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Based on this error message, it seems like that's an artifact of the changes Joe and Megan made in the simularium-viewer recently that are now merged! Seems like a simple change to add that property in `Viewer.tsx`. @interim17 if you could handle that in this PR, that'd be great!
> 
> DIdn't anticipate that because I hadn't update the viewer locally! @ascibisz pushing the change you suggested fixed the mac build but not linux. I'll keep poking around on that one.

Ok! I think I fixed the linux build issue, Linux's case sensitivity in regards to the CSS file names was causing problems. Not sure I fully understand it because we also have mixed case .tsx file names that aren't causing problems, but the build isn't failing now.

File: package-lock.json
Comment (ascibisz) [inline package-lock.json:1]: I'm not a big javascript developer, so this may be a silly question, but why did this file get added now? Is `package-lock.json` redundant with `yarn.lock`?

Comment (interim17) [reply package-lock.json:1]: Honestly I still git a little confused about the role of `package-lock.json` in general and haven't worked on projects that use both node and yarn before. It is supposed to serve as a very specific snapshot of your dependency tree so that you can reproduce things exactly, but it's also common practice to delete it and generate a new one by running `npm i`. Seems like on our team it comes along for the ride with PRs but I worked on another project where we never included it in source control. Wouldn't mind clarification of best practices here from @meganrm

Comment (toloudis) [reply package-lock.json:1]: Usually a project either uses yarn or npm but not both.  yarn.lock is the yarn equivalent of package-lock.json. 
 Generally you are supposed to check in package-lock/yarn.lock.  It represents a working set of current versions of the installed dependencies specified in package.json.  The `npm ci` command uses it to install the exact specific versions of all dependencies (as opposed to what `npm install` does).  It looks like this project is set up to use yarn but I am not sure why - maybe from a third party template project?

Comment (meganrm) [reply package-lock.json:1]: it was set up with yarn, and it was because of the template, but also the reason the template was set up that way is there is a bug (I don't remember what it was) that causes problems with something in the Jupyter setup. This file should be deleted @interim17 and you should always do your installs with yarn and not npm on this projec

Comment (interim17) [reply package-lock.json:1]: ok will do, but to clarify, i should be committing and pushing yarn.lock with this and subsequent PRs

Comment (interim17) [reply package-lock.json:1]: I pushed a commit deleting package-lock and it seems to have broken the windows build, I'll poke around and post here when I figure it out

**Review by ascibisz — APPROVED** [review]
LGTM!

File: package-lock.json
Comment (interim17) [reply package-lock.json:1]: Build issue seems to be resolved, possibly just needed to yarn install for a couple dependencies that were in package-lock and push yarn.lock.

File: src/Viewer.tsx (hunk 1 above)
Comment (meganrm) [reply src/Viewer.tsx]: I would use a valid level from this info: https://www.npmjs.com/package/loglevel

File: src/widget.ts [hunk 2]
```diff
@@ -72,10 +72,13 @@ export class Viewport extends DOMWidgetView {
     const width = this.model.get('width');
     const height = this.model.get('height');
     console.log('rendering viewport', width);
+    const trajectoryData = JSON.parse(this.model.get('trajectory_str'));
+    const trajectoryInfo = trajectoryData.trajectoryInfo;
```
Comment (meganrm) [inline src/widget.ts]: seems out of scope for this PR?

File: css/camera_controls.css [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    flex-direction: column;
+    position: absolute;
+    right: 40px;
+    bottom: 40px;
+    gap: 3px;
+}
+
+.cc-container .btn {
+    border: none;
+    color: #d8d8d8;
```
Comment (meganrm) [inline css/camera_controls.css:12]: maybe another PR, but it's nice to have colors in a single file and use them as variables so they're reusable across components

Comment (interim17) [reply css/camera_controls.css:12]: at first pass this looks like it will take a bit of configuring so i'll make a separate issue for it

File: src/widget.ts (hunk 2 above)
Comment (interim17) [reply src/widget.ts]: yep! removed.
